Index: navit/navit/route.c
===================================================================
--- navit/route.c	(revision 5827)
+++ navit/route.c	(working copy)
@@ -3926,7 +3926,7 @@ int route_remove_attr(struct route *this_, struct attr *attr) {
     dbg(lvl_debug,"enter");
     switch (attr->type) {
     case attr_callback:
-        callback_list_remove(this_->cbl2, attr->u.callback);
+//        callback_list_remove(this_->cbl2, attr->u.callback);
         return 1;
     case attr_vehicle:
         this_->v=NULL;
