Index: navit/CMakeLists.txt
===================================================================
--- navit/CMakeLists.txt	(revision 6084)
+++ navit/CMakeLists.txt	(working copy)
@@ -22,7 +22,7 @@
 set(PACKAGE "navit")
 set(PACKAGE_STRING "${PACKAGE} ${PACKAGE_VERSION}")
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
 include("${PROJECT_SOURCE_DIR}/cmake/navit_macros.cmake")
 
 IF(NOT CMAKE_BUILD_TYPE)
@@ -102,13 +102,13 @@
 add_module(autoload/osso "Default" FALSE)
 add_module(map/garmin "Garmin library not found" FALSE)
 add_feature(USE_NATIVE_LANGUAGE_SUPPORT "Gettext not found and not on Windows" FALSE)
-set(CPACK_PROJECT_CONFIG_FILE "${CMAKE_SOURCE_DIR}/cpack.cmake")
+set(CPACK_PROJECT_CONFIG_FILE "${CMAKE_CURRENT_SOURCE_DIR}/cpack.cmake")
 set(CPACK_PACKAGE_VENDOR "Navit team")
 set(CPACK_PACKAGE_VERSION_MAJOR ${NAVIT_VERSION_MAJOR})
 set(CPACK_PACKAGE_VERSION_MINOR ${NAVIT_VERSION_MINOR})
 set(CPACK_PACKAGE_VERSION_PATCH "${NAVIT_VERSION_PATCH}")
-set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_SOURCE_DIR}/README")
-set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_SOURCE_DIR}/GPL-2")
+set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README")
+set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/GPL-2")
 set(CPACK_NSIS_PACKAGE_NAME "Navit ${PACKAGE_VERSION}")
 set(CPACK_PACKAGE_FILE_NAME ${PACKAGE})
 set(CPACK_PACKAGE_EXECUTABLES "navit;Navit")
@@ -413,7 +413,7 @@
 add_module(map/mg "Default" TRUE)
 endif(NOT MSVC)
 add_module_plugin(support/shapefile "Default" TRUE)
-set(map_shapefile_INCLUDES "${CMAKE_SOURCE_DIR}/navit/support/shapefile")
+set(map_shapefile_INCLUDES "${CMAKE_CURRENT_SOURCE_DIR}/navit/support/shapefile")
 #set(map_shapefile_LIBRARY_DIRS "${CMAKE_BINARY_DIR}/navit/support/shapefile")
 set(map_shapefile_LIBS "support_shapefile")
 add_module(map/shapefile "Default" TRUE)
